lib/astal/io/file.vala

```diff
@@ -85,7 +107,7 @@ public FileMonitor? monitor_file(string path, Closure callback) {
                 if (i.get_file_type() == FileType.DIRECTORY) {
                     var filepath = file.get_child(i.get_name()).get_path();
                     if (filepath != null) {
-                        var m = monitor_file(path, callback);
+                        var m = monitor_file(filepath, callback, follow_symlinks);
```

Comment:
The change from `path` to `filepath` is intentional and seems to be a bug that was never fixed, where monitoring a directory would lead to a infinite recursion (at least the way I understand it).

I can fix it as part of a different pull request if preferred